Fix #20941 : button responsiveness on tablet devices

TARishabh opened the pull request:
## Overview

<!--
READ ME FIRST:
Please answer *all* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #20941 .
2. This PR fixes the issue where the button's display property has been set to none to ensure that it remains hidden and the buttons are already present in the bottom.
3.The original bug occurred because the button was mistakenly included in the navbar component due to incorrect placement in the code.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Proof that changes are correct
Before:
![before-button-responsiveness](https://github.com/user-attachments/assets/b02892a9-eb97-446c-99f0-07bc0ec22936)

After:

[Screencast from 05-10-24 06:45:47 PM IST.webm](https://github.com/user-attachments/assets/652d898b-502f-4661-8274-8881e2981722)


## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

TARishabh committed:
fix: button responsiveness on tablet devices

TARishabh committed:
fix: resolved the two button appearance on desktop screens and overflow of publish button

TARishabh committed:
fix: ran prettier

TARishabh committed:
fix: linting issues

HardikGoyal2003 commented:
@TARishabh There’s a reviewer already assigned to this PR, so I’ll unassign myself. Thanks!

TARishabh commented:
Hello @HardikGoyal2003 ,

I have already formatted the files using Prettier (executing npx prettier . --write and also formatting individual files). However, I am still encountering issues and have opened a discussion regarding this topic.

In the meantime, could you please pull the latest changes from this branch, format the files as necessary, and push the updated branch? Alternatively, if there is a better way for me to resolve this issue myself, I would appreciate your guidance on how to fix it so I can update and push the changes accordingly.

Thank you for your help!

TARishabh commented:
Hey @Vir-8 , PTAL !!
I'm facing some issues with Prettier and have already opened a discussion on it. Could you please run Prettier on the branch and merge it?

TARishabh committed:
Merge branch 'develop' of https://github.com/oppia/oppia into fix-responsive-button-tablet

TARishabh committed:
fix: linting issues

TARishabh committed:
fix: linting issues

TARishabh committed:
fix: linting issues

TARishabh committed:
fix: linting issues

TARishabh committed:
fix: linting issues

TARishabh commented:
Hi @Vir-8 ,

I have resolved the linting issues.

It appears the linting checks were failing because, although the pre-commit hook was running Prettier during commits, it wasn’t formatting the files as expected. I’ve initiated a discussion to investigate the issue further.

Vir-8 commented on core/templates/base-components/base-content.component.css:
```diff
@@ -106,3 +106,13 @@
 }
+
+@media screen and (min-width: 600px) and (max-width: 768px) {
+  .navbar-container .collapse.d-lg-inline-flex.navbar-collapse.oppia-navbar-collapse.ng-cloak.navbar-options-display nav-options {
```
Why are we adding this styling here in the base component? We should not be doing that if the issue exists only in the topic editor

Vir-8 reviewed (changes requested):
Hi @TARishabh, you shouldn't be able to push your changes at all if lint checks fail so are you sure your installation is correct? Also one comment, thanks

TARishabh committed:
fix: created individual css file for topic-editor-navbar and removed related css from base-content-component.css

TARishabh commented:
> Hi @TARishabh, you shouldn't be able to push your changes at all if lint checks fail so are you sure your installation is correct? Also one comment, thanks

It seems I am able to commit and push changes even when the lint checks are failing. I will try reinstalling everything to ensure the setup is correct.

Thank you for your feedback.

TARishabh committed:
fix: linting issues

TARishabh committed:
fix: linting issues

TARishabh commented:
@Vir-8 PTAL !
I have implemented the changes as per your instructions. Specifically, I have created a separate CSS file for the topic navbar and moved the relevant logic from the base-content.component.css to this new individual CSS file.

Please let me know if further adjustments are required.

TARishabh committed:
Merge branch 'develop' into fix-responsive-button-tablet

TARishabh commented:
Hey @Vir-8 PTAL !! Thanks !

Vir-8 commented on core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.css:
```diff
@@ -0,0 +1,9 @@
+@media screen and (min-width: 600px) and (max-width: 768px) {
+  .navbar-container .collapse.d-lg-inline-flex.navbar-collapse.oppia-navbar-collapse.ng-cloak.navbar-options-display nav-options {
```
Why are we selecting the target element with this selector? We generally don't want such ambiguous selectors, could you replace it by a specific class?

Also we shouldn't want to use bootstrap selectors such as d-lg-inline-flex for our custom CSS

HardikGoyal2003 replied:
@TARishabh Please can you answer this?

TARishabh replied:
I observed that two components with similar class names are rendering simultaneously (please see the attachment). Specifically, the components mobile-nav-options and nav-options are both displayed, each with the same classes applied.

To address this, I initially targeted CSS to selectively hide the buttons within a specific screen range, but I believe a more refined approach would be to conditionally render only one of these components at a time based on the window size.

My proposed solution is to update the topic-editor-page.mainpage.html file so that only one of these components—either nav-options or mobile-nav-options—renders depending on the screen size. This would eliminate the need for CSS overrides.

Please confirm if this approach aligns with what you’re looking for so I can proceed with the changes.

@HardikGoyal2003 @Vir-8 


https://github.com/user-attachments/assets/8c8df528-82b5-4dc6-860e-90518e0b0a52

masterboy376 replied:
@HardikGoyal2003 @Vir-8 could you please confirm if the explanation provided by @TARishabh is satisfactory? Thanks!

TARishabh replied:
@HardikGoyal2003   Could you please let me know if there are any changes required on this PR based on the proposed solution? If the explanation and changes are satisfactory, please feel free to resolve the comments. 

Thanks!

Vir-8 reviewed (changes requested):
@TARishabh one comment, also please don't resolve comments on your own, it makes it difficult to keep track. Thanks!

TARishabh commented:
@Vir-8 PTAL !!

cc - @HardikGoyal2003

TARishabh committed:
Merge branch 'develop' of github.com:TARishabh/oppia into fix-responsive-button-tablet

TARishabh committed:
fix: add responsive styles to hide navigation elements on mobile and desktop

TARishabh committed:
Merge branch 'develop' of https://github.com/oppia/oppia into fix-responsive-button-tablet

TARishabh committed:
Revert "fix: linting issues"

This reverts commit b0bcd5e1b4d81a6247fd7a451bb775b3f498f854.

TARishabh committed:
fix: removed custom css class and added media queries on topic-editor-mainpage

TARishabh committed:
fix: removed topic-editor-navbar.component.css from topic-editor-navbar.component.ts

TARishabh committed:
Merge branch 'fix-responsive-button-tablet' of github.com:TARishabh/oppia into fix-responsive-button-tablet

TARishabh commented:
@Vir-8 PTAL !!

mon4our commented:
@TARishabh can you show how the warning is displayed in tablet mode after your changes?

TARishabh commented:
@mon4our can you elaborate which warnings?

mon4our commented:
@TARishabh 
![image](https://github.com/user-attachments/assets/ca0f9bdd-db62-4d9c-86ff-86f15a6c8d82)
On pc when you hover over the yellow caution sign, you see a warning message. So I want to see how that message is displayed on tablet mode after your changes.

TARishabh commented:
@mon4our Here it is:
![Screenshot from 2024-11-11 01-06-00](https://github.com/user-attachments/assets/64d7a905-668a-4bc9-9ac4-68ea3671756a)
![Screenshot from 2024-11-11 01-05-52](https://github.com/user-attachments/assets/b5f51ec1-068e-4be8-a13d-a205e03dabf7)

mon4our commented:
@TARishabh this looks great! Can you also show a video recording of this? Thanks!

mon4our commented:
Can you also check the failing CI checks?
The ui part LGTM!

Ash-2k3 commented:
Hi @Vir-8, can you PTAL at this PR, Thanks!

TARishabh commented:
Hey @Ash-2k3 , I looked into the failing CI checks. It seems the checks are failing because the username input is not visible during the test. I'm not sure if this is just a flaky test or if it's a specific issue with this PR. Could you please guide me on how to proceed?

Thanks!

![image](https://github.com/user-attachments/assets/2e3e7cd6-9d98-497b-b006-22924a54722d)

TARishabh committed:
Merge branch 'develop' into fix-responsive-button-tablet

TARishabh commented:
@Ash-2k3 PTAL !

cc - @HardikGoyal2003 @seanlip

seanlip commented:
@TARishabh You need to make sure the Assignees section in the right sidebar has the correct folks if you want them to take a look at your PR.

TARishabh commented:
Hi @seanlip , Apologies for the oversight. Please let me know if there’s anything else I need to address on this PR. Thanks !

seanlip committed:
Merge branch 'develop' into fix-responsive-button-tablet

seanlip commented:
@TARishabh Some of your CI tests seem to be failing repeatedly.  I merged develop into your branch, please let us know if it still happens. Thanks.

Also, here is how to go about this in general: if you see a CI failure, then either it's due to your PR or it isn't. If it isn't, find the issue that is already filed for it, add a comment to that issue, and link to that issue from here (and request a restart of the relevant test). This is the easiest way to get your PR through the checks. Thanks!

TARishabh commented:
@seanlip ,I've reviewed the two failing CI checks:

1. `acceptance_test`: It seems to be failing due to a timeout while waiting for the "button.e2e-test-dismiss-welcome-modal" and "Completed goals section is empty." This does not appear to be related to my changes, which focus on button responsiveness on tablet devices.

2. `e2e_lighthouse_performance_acceptance_workflow`: This also seems unrelated, as it appears to be a performance check issue that doesn’t interact with the UI changes I've made.

I believe these failures are unrelated to my PR. I will now:
1. Find if there are any open issues for these failures.
2. If an issue exists, I will link it here and add a comment to that issue indicating my PR is blocked.
3. If no issue exists, I will open a new issue.

Thanks, and please let me know if there's anything else I should do.

seanlip commented:
@TARishabh Thanks. Don't worry about the second check, it fails if any of the individual ones fail.

Have you left a comment on the issue for the 1st?
